public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -58,155 +62,183 @@ const RulesFilter = () => {
   }));
 
   const handleDataSourceChange = (dataSourceValue: DataSourceInstanceSettings) => {
-    setQueryParams({ dataSource: dataSourceValue.name });
+    updateFilters({ ...filterState, dataSourceName: dataSourceValue.name });
+    setFilterKey((key) => key + 1);
   };
 
   const clearDataSource = () => {
-    setQueryParams({ dataSource: null });
+    updateFilters({ ...filterState, dataSourceName: undefined });
+    setFilterKey((key) => key + 1);
   };
 
   const handleQueryStringChange = debounce((e: FormEvent<HTMLInputElement>) => {
     const target = e.target as HTMLInputElement;
-    setQueryParams({ queryString: target.value || null });
+    setSearchQuery(target.value);
   }, 600);
 
-  const handleAlertStateChange = (value: string) => {
+  const handleAlertStateChange = (value: PromAlertingRuleState) => {
     logInfo(LogMessages.clickingAlertStateFilters);
-    setQueryParams({ alertState: value });
+    updateFilters({ ...filterState, ruleState: value });
+    setFilterKey((key) => key + 1);
   };
 
   const handleViewChange = (view: string) => {
     setQueryParams({ view });
   };
 
   const handleRuleTypeChange = (ruleType: PromRuleType) => {
-    setQueryParams({ ruleType });
+    updateFilters({ ...filterState, ruleType });
+    setFilterKey((key) => key + 1);
   };
 
   const handleClearFiltersClick = () => {
-    setQueryParams({
-      alertState: null,
-      queryString: null,
-      dataSource: null,
-      ruleType: null,
-    });
+    setSearchQuery(undefined);
+
+    if (onFilterCleared) {
```

Comment:
Perhaps we should default this to a `noop` instead of checking if it's defined?

Reply:
It's a prop for an additional action that you may want to invoke after clearing out filters not to clear the filter expression